internal/pkg/template/templates/workloads/partials/cf/taskrole.yml

```diff
@@ -2,9 +2,18 @@ TaskRole:
   Metadata:
     'aws:copilot:description': 'An IAM role to control permissions for the containers in your tasks'
   Type: AWS::IAM::Role
-  Properties:{{if .NestedStack}}{{$stackName := .NestedStack.StackName}}{{if gt (len .NestedStack.PolicyOutputs) 0}}
-    ManagedPolicyArns:{{range $managedPolicy := .NestedStack.PolicyOutputs}}
-    - Fn::GetAtt: [{{$stackName}}, Outputs.{{$managedPolicy}}]{{end}}{{end}}{{end}}
+  Properties:
+    {{- if or (.NestedStack) (eq .Observability.Tracing "AWSXRAY")}}
```

Comment:
We should create a `ManagedPolicyArns` field only if the `NestedStack` has `PolicyOutputs` right now it's possible that we render a template like this:
```
Properties:
  ManagedPolicyArns:
  AssumeRolePolicyDocument:
```
if there are addons and no managed policy and tracing afaict

Reply:
good catch! fixed 😊